tests/unit/BitbucketAPI.test.ts

```diff
@@ -85,10 +68,13 @@ describe('mergePullRequest', () => {
       .mockResolvedValueOnce({ data: { task_status: 'PENDING' } })
       .mockResolvedValueOnce({ data: { task_status: 'PENDING' } })
       .mockRejectedValueOnce({ response: {} });
-    await expect(mergePullRequest(landRequestStatus)).rejects.toThrow();
-    expect(loggerInfoSpy).toHaveBeenCalledTimes(2);
-    expect(loggerErrorSpy).toHaveBeenCalledTimes(1);
-    expect(mockedDelay).toHaveBeenCalledTimes(2);
+    expect(await mergePullRequest(landRequestStatus)).toEqual(BitbucketAPI.FAILED);
+  });
+
+  test('Timeout merge polling', async () => {
```

Comment:
are you testing the counter as well?

Reply:
This is testing the max attempts, but I couldn't figure out how to test the abort thing, because it's a function that's called while the async merging is occurring.

Reply:
You might be able to do it by using normal promises rather than async/await.

```
const mergePromise = mergePullRequest(landRequestStatus).then(res => { expect(res).toEqual(BitbucketAPI.ABORT); });
bitbucketAPI.cancelMergePolling(...);
return mergePromise;
```

Could also return the `done` callback parameter instead of returning the promise as well - https://jestjs.io/docs/asynchronous

Reply:
Yeah I attempted this and it didn't work. Will merge for now and potentially come back to that later.